nifi-commons/nifi-security-identity/src/main/java/org/apache/nifi/authorization/util/IdentityMappingUtil.java

```diff
@@ -140,8 +141,12 @@ public int compare(IdentityMapping m1, IdentityMapping m2) {
      * @return the mapped identity, or the same identity if no mappings matched
      */
     public static String mapIdentity(final String identity, List<IdentityMapping> mappings) {
-        for (IdentityMapping mapping : mappings) {
-            Matcher m = mapping.getPattern().matcher(identity);
+        final List<String> keys = mappings.stream().map(m -> m.getKey()).collect(Collectors.toList());
+        Collections.sort(keys);
+
+        for (final String key : keys) {
+            final IdentityMapping mapping = mappings.stream().filter(m -> m.getKey().equals(key)).findFirst().orElseThrow();
```

Comment:
Instead of iterating over the keys and then searching for the matching IdentityMapping, did you consider creating a `TreeMap` of the key and the `IdentityMapping` object? That would avoid the need for this stream, filter, and findFirst approach.

Reply:
I did not think of this, but a good idea. For the whole picture, I was hesitant with the change because while the `TreeMap` avoids the constructs above (which in this case as we are not talking about bigger data structures, might not be have as much affect on performance, using an `entrySet` [1], where the contract would allow too much freedom on the order of the elements (even if the implementation provides further restrictions) seems like a questionable move forward. In the end, after weighting the implications, I decided to go with the `TreeMap`. I will commit the changes soonish.

[1] using the `keySet` would result the same, and the `forEach` looks to be a bit convoluted here.